File: src/infrastructure/sqlite/mappers/memberMapper.ts
import type { Member } from "@/src/domain/members/Member";

import type { MemberRow } from "../rows/MemberRow";

export function mapMemberRow(row: MemberRow): Member {
  return {
    id: row.id,
    ownerUserId: row.owner_user_id,
    displayName: row.display_name,
    createdAt: row.created_at,
    updatedAt: row.updated_at,
  };
}

File: backend/src/main/java/com/debtulator/backend/members/MemberMapper.java
package com.debtulator.backend.members;

import org.springframework.stereotype.Component;

import java.util.LinkedHashMap;
import java.util.Map;

@Component
public class MemberMapper {

    public Map<String, Object> toSyncPayload(Member member) {
        Map<String, Object> payload = new LinkedHashMap<>();

        payload.put("id", member.getId().toString());
        payload.put("ownerUserId", member.getOwnerUserId().toString());
        payload.put("displayName", member.getDisplayName());
        payload.put(
                "linkedUserId",
                member.getLinkedUserId() != null
                        ? member.getLinkedUserId().toString()
                        : null
        );
        payload.put("createdAt", member.getCreatedAt().toString());
        payload.put("updatedAt", member.getUpdatedAt().toString());
        payload.put("version", member.getVersion());

        return payload;
    }
}
